opensrp-reveal/src/main/java/org/smartregister/reveal/util/NativeFormProcessor.java

```diff
@@ -0,0 +1,269 @@
+        allSharedPreferences = CoreLibrary.getInstance().context().allSharedPreferences();
+        gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+                .registerTypeAdapter(DateTime.class, new DateTimeTypeConverter()).create();
+    }
+
+    public NativeFormProcessor(JSONObject jsonObject) {
+        this.jsonForm = jsonObject;
+        allSharedPreferences = CoreLibrary.getInstance().context().allSharedPreferences();
+        gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+                .registerTypeAdapter(DateTime.class, new DateTimeTypeConverter()).create();
+    }
+
+    public NativeFormProcessor withEntityId(String entityId) {
+        this.entityId = entityId;
+        return this;
+    }
+
+    public NativeFormProcessor withEncounterType(String encounterType) {
+        this.encounterType = encounterType;
+        return this;
+    }
+
+    public NativeFormProcessor withBindType(String bindType) {
+        this.bindType = bindType;
+        return this;
+    }
+
+    public NativeFormProcessor hasClient(boolean hasClient) {
+        this.hasClient = hasClient;
+        return this;
+    }
+
+    private JSONObject getOrCreateDetailsNode() throws JSONException {
+        JSONObject formData = jsonForm.has(DETAILS) ? jsonForm.getJSONObject(DETAILS) : null;
+        if (formData == null) {
+            formData = new JSONObject();
+            formData.put(Constants.Properties.FORM_VERSION, jsonForm.optString("form_version"));
+            formData.put(Constants.Properties.APP_VERSION_NAME, BuildConfig.VERSION_NAME);
+
+            jsonForm.put(DETAILS, formData);
+        }
+
+        return formData;
+    }
+
+    public NativeFormProcessor tagTaskDetails(Task task) throws JSONException {
+        JSONObject formData = getOrCreateDetailsNode();
+        formData.put(Constants.Properties.TASK_IDENTIFIER, task.getIdentifier());
+        formData.put(Constants.Properties.TASK_BUSINESS_STATUS, task.getBusinessStatus());
+        formData.put(Constants.Properties.TASK_STATUS, task.getStatus());
+        return this;
+    }
+
+    public NativeFormProcessor tagLocationData(Location location) throws JSONException {
+        JSONObject formData = getOrCreateDetailsNode();
+        formData.put(Constants.Properties.LOCATION_ID, location.getId());
+        formData.put(Constants.Properties.LOCATION_UUID, location.getId());
+        formData.put(Constants.Properties.LOCATION_VERSION, location.getServerVersion().toString());
+        return this;
+    }
+
+    private JSONArray getFields() throws JSONException {
+        if (_fields == null)
+            _fields = jsonForm.getJSONObject(JsonFormConstants.STEP1).getJSONArray(JsonFormConstants.FIELDS);
+
+        return _fields;
+    }
+
+    private Client createClient() throws JSONException {
+        if (hasClient && _client == null)
+            _client = org.smartregister.util.JsonFormUtils.createBaseClient(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'org.smartregister.util.JsonFormUtils.createBaseClient' due to existing import 'org.smartregister.util.JsonFormUtils'](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5414138)